sagemaker/data_generation.ipynb

```diff
@@ -0,0 +1,142 @@
+    "import pandas as pd\n",
+    "from io import StringIO\n",
+    "import io,random,datetime,boto3,sagemaker,os,time,json"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "os.environ[\"AWS_ACCESS_KEY_ID\"] = ACCESS_KEY\n",
+    "os.environ[\"AWS_SECRET_ACCESS_KEY\"] = SECRET_KEY\n",
+    "os.environ[\"AWS_DEFAULT_REGION\"] = REGION_NAME\n",
+    "\n",
+    "boto_session = boto3.session.Session(aws_access_key_id=ACCESS_KEY,aws_secret_access_key=SECRET_KEY,region_name=REGION_NAME)\n",
+    "sagemaker_session = sagemaker.Session(boto_session)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def save_dataset_to_s3(data, monitor_name, name, typeofdata):\n",
+    "    file_name = name + \".csv\"\n",
+    "    data_dir = os.path.join(\n",
+    "        monitor_name,\n",
+    "        typeofdata\n",
+    "    )\n",
+    "    file_path = os.path.join(data_dir, file_name)\n",
+    "    with io.StringIO() as csv_buffer:\n",
+    "        data.to_csv(csv_buffer, index=False)\n",
+    "\n",
+    "        response =  boto3.client(\"s3\").put_object(\n",
+    "            Bucket=BUCKET, Key=file_path, Body=csv_buffer.getvalue()\n",
+    "        )\n",
+    "        status = response.get(\"ResponseMetadata\", {}).get(\"HTTPStatusCode\")\n",
+    "\n",
+    "        if status == 200:\n",
+    "            print(f\"Successful S3 put_object response. Status - {status}\")\n",
+    "            print(file_path)\n",
+    "        else:\n",
+    "            print(f\"Unsuccessful S3 put_object response. Status - {status}\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Invoke Endpoint"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "count = 1\n",
+    "while True:\n",
+    "    dataframe = pd.read_csv('insurance.csv')\n",
+    "    dataframe = dataframe.iloc[1:,0:7]\n",
+    "    no_of_samples = random.randint(10,15)\n",
+    "    sample_df = dataframe.sample(no_of_samples)\n",
+    "    labels = sample_df[\"charges\"]\n",
+    "    sample_df = sample_df.drop([\"charges\"],axis=1)\n",
+    "    csv_file = io.StringIO()\n",
+    "    # by default sagemaker expects comma seperated\n",
+    "    sample_df.to_csv(csv_file, sep=\",\", header=False, index=False)\n",
+    "    my_payload_as_csv = csv_file.getvalue()\n",
+    "    print(\"Invoking endpoint....\")\n",
+    "    predictor = sagemaker.predictor.Predictor(\n",
+    "               endpoint_name, \n",
+    "               sagemaker_session=sagemaker_session,\n",
+    "               serializer=sagemaker.serializers.CSVSerializer(),\n",
+    "               ContentType=\"text/csv\")                                                           \n",
+    "    response=predictor.predict(my_payload_as_csv)\n",
```

Comment:
Doing batch prediction?
And, this goes forever? Can we put a limit like an hr or so that they can define upfront in some cell?